src/dotty/tools/dotc/core/pickling/TreePickler.scala

```diff
@@ -531,8 +535,16 @@ class TreePickler(pickler: TastyPickler) {
       withLength { pickleType(ann.symbol.typeRef); pickleTree(ann.tree) }
     }
 
+    def updateMapWithDeltas[T](mp: collection.mutable.Map[T, Addr]) = {
+      mp.map{
+        case (key, addr) => (key, adjusted(addr))
```

Comment:
@retronym: Good point. We could modify like this

    for (key <- mp.keysIterator.toBuffer) mp(key) = adjusted(mp(key))

That looks like the most efficient safe solution. I believe efficiency is a concern here - these maps can become large.